sparkle/solver/solver.py

```diff
@@ -59,11 +60,40 @@ def get_pcs_file(self: Solver) -> Path:
                 file_count += 1
 
         if file_count != 1:
+            return False
+
+        return self.directory / file_name
+
+    def check_pcs_file_exists(self: Solver) -> bool:
+        """Check if the parameter file exists.
+
+        Returns:
+            Boolean if there is one pcs file in the solver directory.
+        """
+        return isinstance(self._get_pcs_file(), Path)
+
+    def get_pcs_file(self: Solver) -> Path:
+        """Get path of the parameter file.
+
+        Returns:
+            Path to the parameter file.
+        """
+        if not (file_path := self._get_pcs_file()):
             print("None or multiple .pcs files found. Solver "
                   "is not valid for configuration.")
             sys.exit(-1)
 
-        return self.directory / file_name
+        return file_path
+
+    def read_pcs_file(self: Solver) -> None:
+        """Read the pcs file."""
+        pcs_file = self._get_pcs_file()
+        try:
+            parser = pcsparser.PCSParser()
+            parser.load(str(pcs_file), convention="smac")
+            print("Could read pcs file")
```

Comment:
Redundant?

Reply:
Not redundant, but can move printing to CLI/add_solver.py and change to function return to a bool

Reply:
Sounds good to me